controllers/fg_handler.go

```diff
@@ -0,0 +1,52 @@
+package controllers
+
+import (
+	"context"
+
+	"github.com/openshift/sandboxed-containers-operator/internal/featuregates"
+)
+
+// Create two enums to represent the state of the feature gates
+type FeatureGateState int
+
+const (
+	Enabled FeatureGateState = iota
+	Disabled
+)
+
+// Function to handle the feature gates
+func (r *KataConfigOpenShiftReconciler) processFeatureGates() error {
+
+	if r.FeatureGates == nil {
+		return nil
+	}
+
+	// Check which feature gates are enabled in the FG ConfigMap and
+	// perform the necessary actions
+	// The feature gates are defined in internal/featuregates/featuregates.go
+	// and are fetched from the ConfigMap in the namespace
+	// Eg. TimeTravelFeatureGate
+
+	if r.FeatureGates.IsEnabled(context.TODO(), featuregates.TimeTravelFeatureGate) {
+		r.Log.Info("Feature gate is enabled", "featuregate", featuregates.TimeTravelFeatureGate)
+		// Perform the necessary actions
+		r.handleTimeTravelFeature(Enabled)
+	} else {
+		r.Log.Info("Feature gate is disabled", "featuregate", featuregates.TimeTravelFeatureGate)
+		// Perform the necessary actions
+		r.handleTimeTravelFeature(Disabled)
+	}
+
+	return nil
+
+}
+
+// Function to handle the TimeTravel feature gate
+func (r *KataConfigOpenShiftReconciler) handleTimeTravelFeature(state FeatureGateState) {
```

Comment:
nit that is not a problem for now, but maybe for the future: maybe move this to a specific file, so we can isolate code from different features.

Reply:
You mean feature specific file right ?

Reply:
yes.